DigitalLearningSolutions.Data/Services/GroupsService.cs

```diff
@@ -223,21 +225,44 @@ public void DeleteDelegateGroup(int groupId, bool deleteStartedEnrolment, DateTi
             transaction.Complete();
         }
 
-        public GroupCourse? GetGroupCourse(int groupCustomisationId, int groupId, int centreId)
-        {
-            return groupsDataService.GetGroupCourse(groupCustomisationId, groupId, centreId);
-        }
-
-        public void RemoveGroupCourseAndRelatedProgress(int groupCustomisationId, int groupId, bool deleteStartedEnrolment)
+        public void RemoveGroupCourseAndRelatedProgress(
+            int groupCustomisationId,
+            int groupId,
+            bool deleteStartedEnrolment
+        )
         {
             using var transaction = new TransactionScope();
 
-            groupsDataService.RemoveRelatedProgressRecordsForCourse(groupId, groupCustomisationId, deleteStartedEnrolment, clockService.UtcNow);
+            groupsDataService.RemoveRelatedProgressRecordsForGroupCourse(
+                groupId,
+                groupCustomisationId,
+                deleteStartedEnrolment,
+                clockService.UtcNow
+            );
             groupsDataService.DeleteGroupCustomisation(groupCustomisationId);
 
             transaction.Complete();
         }
 
+        public GroupCourse? GetActiveGroupCourse(int groupCustomisationId, int groupId, int centreId)
+        {
+            var groupCourse = groupsDataService.GetGroupCourse(groupCustomisationId, groupId, centreId);
+
+            if (groupCourse == null || !groupCourse.Active || groupCourse.InactivatedDate != null ||

```

Comment:
It would be cool to put this predicate on the GroupCourse object so we can give it a nice name like "IsCurrentCourse" and we have some encoded logic... but maybe with a better name. I'm trying to avoid "IsActiveCourse" because them you get a meaning clash with `IsActive`. Any ideas?

Reply:
Minor point while I remember: I'd disagree with the name "IsCurrentCourse" simply because we have the "Current" activites page in the learning portal that displays various courses. I can't imagine we'll be making many changes to that page soon (or at all) but I think the name could potentially conflate with that concept.

I think a better name would be "IsCourseAccessible" as those flags preclude a course from being enrolled on, but then that might have some overlap with where we check if an admin can access courses (or even if a centre has a course). I think with the recent renaming of the ...CanAccessCourse to ...CanManageCourse and ...CanViewCourse, it might be acceptable.

Reply:
Following on from a previous comment I've gone with `IsUsable`.
Additionally the method names have changed to reflect this: `GetUsableGroupCourseForCentre`

Reply:
Good choice!